src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -733,14 +733,26 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                     // We're awaiting the task to propagate the exception in this case.
                     if (Volatile.Read(ref _disposed) == Status_Disposed)
                     {
-                        if (async)
+                        try
                         {
-                            await sendRequestContentTask.ConfigureAwait(false);
+                            if (async)
+                            {
+                                await sendRequestContentTask.ConfigureAwait(false);
+                            }
+                            else
+                            {
+                                // No way around it here if we want to get the exception from the task.
+                                sendRequestContentTask.GetAwaiter().GetResult();
+                            }
                         }
-                        else
+                        catch (Exception ex)
                         {
-                            // No way around it here if we want to get the exception from the task.
-                            sendRequestContentTask.GetAwaiter().GetResult();
+                            // Map the exception the same way as we normally do.
+                            if (MapSendException(ex, cancellationToken, out Exception mappedEx))
+                            {
+                                throw mappedEx;
+                            }
+                            throw;
```

Comment:
This could be smaller and avoid a rethrow:

```c#
catch(Exception ex) when(MapSendException(ex, cancellationToken, out Exception mappedEx))
{
   throw mappedEx;
}
```